src/monobase/pget.py

```diff
@@ -123,6 +124,25 @@ def single_pget(url: str, dest: str, extract: bool, force: bool) -> None:
     if not force:
         assert not os.path.exists(dest)
 
+    m = hashlib.sha256()
+    m.update(url.encode())  # default encoding of utf-8
+    fpath = os.path.join(KNOWN_WEIGHTS_DIR, m.hexdigest())
+    if os.path.exists(fpath):
+        if extract:
+            # rm -rf if -xf
+            if force:
+                os.rmdir(fpath)
```

Comment:
Sorry I got it wrong. `pget -xf <tar> <dir>` overwrites `<dir>` if it exists, but doesn't delete files in there not present in `<tar>`. So we don't need to delete it really. Also `os.rmdir` doesn't recursively delete so this might not work anyway.

OTOH if `dest` is a file, both `pget -xf` and `tar -xf -C` fails so their behavior match.